src/loaders/sdf.cpp

```diff
@@ -0,0 +1,211 @@
+{
+
+// ----------------------------------------------------------------------------------------------------
+
+static const std::set<std::string> SDF_ARRAY_SET {"include", "link" , "collision", "visual", "joint",
+                                                 "shape", "areas"}; //temp for development.
+static const std::set<std::string> SDF_MAP_SET {"sdf", "world", "audio", "wind", "pose", "atmosphere", "gui", "camera",
+                                          "track_visual", "plugin", "model", "gripper", "geometry", "box",
+                                          "cylinder", "heightmap", "blend", "image", "mesh", "texture", "plane",
+                                          "polyline", "sphere" };
+static const std::set<std::string> SDF_VALUE_SET {"device", "linear_velocity", "uri", "name", "static", "pose", "gravity",
+                                            "magnetic_field", "type", "temperature", "pressure", "temperature_gradient",
+                                            "fullscreen", "view_controller", "projection_type", "min_dist", "max_dist",
+                                            "use_model_frame", "xyz", "inherit_yaw", "filename", "empty", "size",
+                                            "radius", "length", "pos", "diffuse", "normal", "min_height", "fade_dist",
+                                            "sampling", "point", "height" };
+
+
+// ----------------------------------------------------------------------------------------------------
+
+tue::config::NodeType getSDFNodeType (const std::string& element_name)
+{
+    std::set<std::string>::const_iterator it = SDF_ARRAY_SET.find(element_name);
+    if (it != SDF_ARRAY_SET.end())
+        return ARRAY;
+
+    it = SDF_MAP_SET.find(element_name);
+    if (it == SDF_MAP_SET.end())
+        std::cout << "Element: '" << element_name << "' not in SDF ARRAY or MAP list. Will return MAP as type." << std::endl;
+
+    return MAP;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool loadFromSDFElement(const TiXmlElement& element, ReaderWriter& config, const NodeType node_type)
+{
+    if (element.FirstChildElement() == nullptr)
+    {
+        return loadFromXMLText(element, config);
+    }
+    else
+    {
+        // Start new group or array item
+        // in case of array the array is opened on parent level. Only need to add array items on this level.
+        if (node_type == ARRAY)
+            config.addArrayItem();
+        else
+            // Start a new group
+            config.writeGroup(element.ValueStr());
+
+
+        // Iterate through attributes
+        // ToDo: this does not work if this element does not contain children (we don't end up here)
+        for (const TiXmlAttribute* attribute = element.FirstAttribute(); attribute != nullptr; attribute = attribute->Next())
+        {
+            int i;
+            if (attribute->QueryIntValue(&i) == TIXML_SUCCESS)
```

Comment:
What happens with 1.0? Is this correctly passed as a double? In other words: does the order of these statements matter?

Reply:
I will implement this in a test. The easiest way to check.

Reply:
I didn’t work correctly. I replaced it with the code already used for element values.